Added stream functions

harrysinghtiwana opened the pull request:
I have added some more useful methods to enhance the API functionalities for the followings;

Added method getDeltaCursor() to fetch the latest delta cursors
Added method getDeltas() to fetch the deltas as per cursor in raw format.
Added method getCalenders() to fetch the calendars.
Added method getDeltaStream() to fetch the detlas in an array as per include types.
Added method getContacts() to fetch the contacts.
Added reusable methods such as setApiClientOptions() and createAuthHeader() as well.

Commit:
Added functions for Deltas 

Added function getDeltas, getDeltaCursor and createAuthHeader

Commit:
Fixed the braces as required by Nylas

Commit:
Removed namespace

Commit:
Merge remote-tracking branch 'nylas/master'

Commit:
Removed namespace for delta cursor and deltas

harrysinghtiwana committed:
Added stream functions to enhance the API.

duxet commented:
Hey, you might want to take a look at my fork: https://github.com/O323/nylas-php

harrysinghtiwana committed:
Added custom written curl function due to Guzzle connectivity issue with Nylas API. It will be improved latter on properly.

harrysinghtiwana committed:
Improved get contacts method to return list of filtered emails with name
Added another temporary function to get calendar events results by native curl request.
